spec/broadside/configuration_spec.rb

```diff
@@ -25,18 +25,61 @@
     expect(Broadside.config.valid?).to be false
   end
 
-  context 'ssh' do
+  describe '#ssh_cmd' do
     let(:ip) { '123.123.123.123' }
+    let(:user) { 'test-user' }
+    let(:keyfile) { 'path_to_keyfile' }
+    let(:proxy_user) { 'proxy-user' }
+    let(:proxy_host) { 'proxy-host' }
+    let(:proxy_port) { '22' }
+    let(:proxy_keyfile) { 'path_to_proxy_keyfile' }
+    let(:ssh_config) { {} }
 
-    it 'should build the ssh command with a user' do
-      expect(Broadside.config.ssh_cmd(ip)).to eq("ssh -o StrictHostKeyChecking=no #{user}@#{ip}")
+    before(:each) do
+      Broadside.config.ssh = ssh_config
     end
 
-    it 'should build the ssh command without a user' do
-      Broadside.configure do |config|
-        config.ssh = {}
-      end
+    it 'should build the ssh command' do
       expect(Broadside.config.ssh_cmd(ip)).to eq("ssh -o StrictHostKeyChecking=no #{ip}")
     end
+
+    context 'with tty option' do
+      it 'generates an ssh command string with -tt flags' do
+        expect(Broadside.config.ssh_cmd(ip, tty: true)).to eq(
+          "ssh -o StrictHostKeyChecking=no -t -t #{ip}"
+        )
+      end
+    end
+
+    context 'with ssh user and keyfile provided in the config' do
+      let(:ssh_config) { { user: user, keyfile: keyfile } }
+
+      it 'generates an ssh command string with keyfile flag and user set' do
+        expect(Broadside.config.ssh_cmd(ip)).to eq(
+          "ssh -o StrictHostKeyChecking=no -i #{keyfile} #{user}@#{ip}"
+        )
+      end
+    end
+
+    context 'with ssh proxy' do
```

Comment:
that makes it clear that proxy config is just a subcontext of configuring ssh options, which it is.